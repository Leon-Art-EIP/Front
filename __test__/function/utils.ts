import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
