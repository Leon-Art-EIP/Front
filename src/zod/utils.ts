export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
