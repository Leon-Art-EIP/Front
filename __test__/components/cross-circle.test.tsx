import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import CrossCircle from "../../src/components/animated/cross-circle";

test("CrossCircle snapshot", () => {
  testSnapshot(<CrossCircle />);
});
