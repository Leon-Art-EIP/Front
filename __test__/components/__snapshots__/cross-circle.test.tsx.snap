// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CrossCircle snapshot 1`] = `
<svg
  className="cancel-icon"
  viewBox="0 0 52 52"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    className="cancel-circle"
    cx="26"
    cy="26"
    fill="none"
    r="25"
  />
  <path
    className="cancel-cross"
    d="M16 16, 36 36, M36 16, 16 36"
    fill="none"
  />
</svg>
`;
